sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodShouldNotOnlyReturnConstant.cs

```diff
@@ -46,7 +47,7 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var methodDeclaration = (MethodDeclarationSyntax)c.Node;
 
-                    if (methodDeclaration.ParameterList?.Parameters.Count > 0)
+                    if (methodDeclaration.ParameterList?.Parameters.Count > 0 || IsVirtual(methodDeclaration))
```

Comment:
Usually we format all additional conditions on new lines:
```
    if (methodDeclaration.ParameterList?.Parameters.Count > 0 ||
        IsVirtual(methodDeclaration))
```

Reply:
On the other hand, for partial methods you could be raising on the parts that do not have `virtual` keyword (I did not check but have the feeling it is possible). We already have the symbol of the method, so it could be safer to use it instead of the syntax.

Reply:
As discussed offline, [partial methods](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/partial-method) must be void, so the rule does not apply